### PR TITLE
Bugfix freeze on Linux

### PR DESCRIPTION
**Pull request by haesleinhuepf**
Hi everyone,

here comes a bugfixe related to https://github.com/imagej/imagej-legacy/issues/227

Instead of submitting individual completions to the AutoComplete (upstream dependency), we send a list of completions. This prevents the list of completions being sorted again and again. The issue appeared on some Linux/MacOS computers not very reproducibly. However, on @frauzufall laptop we were able to reproduce it and fix it.

Let me know if you see any issue. I would otherwise merge in 7 days.

Cheers,
Robert

### CODE REVIEW

File: src/main/java/net/imagej/legacy/plugin/MacroAutoCompletionProvider.java [hunk 1]
```diff
@@ -153,16 +153,19 @@ void addModuleCompletions(ModuleService moduleService) {
 		sorted = false;
 		this.moduleService = moduleService;
 
+		ArrayList<Completion> list = new ArrayList<Completion>();
```
Comment (imagejan) [inline src/main/java/net/imagej/legacy/plugin/MacroAutoCompletionProvider.java]: Can we name this more concisely `completions`, as in the `addMacroExtensionAutoCompletions` method? Also, the redundant type arguments can be removed from the constructor call:

```suggestion
		ArrayList<Completion> completions = new ArrayList<>();
```

File: src/main/java/net/imagej/legacy/plugin/MacroAutoCompletionProvider.java [hunk 2]
```diff
@@ -153,16 +153,19 @@ void addModuleCompletions(ModuleService moduleService) {
...
+
 		for (ModuleInfo info : moduleService.getModules()) {
 			if(info.getMenuPath().getLeaf() != null) {
 				String name = info.getMenuPath().getLeaf().getName().trim();
 				String headline = "run(\"" + name +"\")";
 				String description = "<b>" + headline + "</b><p>" +
 						"<a href=\"https://imagej.net/Special:Search/" + name.replace(" ", "%20") + "\">Search imagej wiki for help</a>";
 
-				addCompletion(makeListEntry(this, headline, null, description));
+				list.add(makeListEntry(this, headline, null, description));
```
Comment (imagejan) [inline src/main/java/net/imagej/legacy/plugin/MacroAutoCompletionProvider.java]:
```suggestion
				completions.add(makeListEntry(this, headline, null, description));
```

(see above)

File: src/main/java/net/imagej/legacy/plugin/MacroAutoCompletionProvider.java [hunk 3]
```diff
@@ -153,16 +153,19 @@ void addModuleCompletions(ModuleService moduleService) {
...
 			}
 		}
+		addCompletions(list);
```
Comment (imagejan) [inline src/main/java/net/imagej/legacy/plugin/MacroAutoCompletionProvider.java]:
```suggestion
		addCompletions(completions);
```
(see above)

File: src/main/java/net/imagej/legacy/plugin/MacroAutoCompletionProvider.java [hunk 4]
```diff
@@ -173,9 +176,12 @@ public void addMacroExtensionAutoCompletions(MacroExtensionAutoCompletionService
 		this.macroExtensionAutoCompletionService = macroExtensionAutoCompletionService;
 
 		List<BasicCompletion> completions = macroExtensionAutoCompletionService.getCompletions(this);
+		List<Completion> completionsCopy = new ArrayList<Completion>();
```
Comment (imagejan) [inline src/main/java/net/imagej/legacy/plugin/MacroAutoCompletionProvider.java]: Remove redundant type arguments:
```suggestion
		List<Completion> completionsCopy = new ArrayList<>();
```

**Comment by haesleinhuepf** [comment]
Cool, thanks @imagejan ! First time I commited changes via Browser in this view here :-)

**Pull request merged**